pipeline.py

```diff
@@ -84,19 +99,49 @@ def _process(self, raw_text: str):
         """
         Processes each token and creates MorphToken class instance
         """
-        pass
+        # cleaned_text = ' '.join(re.findall(r'[а-яёА-ЯЁ]+', raw_text))
+        # analyzed_cleaned_text = Mystem().analyze(cleaned_text)
 
 
 def validate_dataset(path_to_validate):
     """
     Validates folder with assets
     """
-    pass
+    path_to_dataset = Path(path_to_validate)
+
+    if not path_to_dataset.exists():
+        raise FileNotFoundError
+
+    if not path_to_dataset.is_dir():
+        raise NotADirectoryError
+
+    list_of_txts = list(path_to_dataset.glob('*_raw.txt'))
+    list_of_jsons = list(path_to_dataset.glob('*_meta.json'))
+
+    if not len(list_of_txts) == len(list_of_jsons):
+        raise InconsistentDatasetError
+
+    txt_indices = []
+    json_indices = []
+    for txt, json in zip(list_of_txts, list_of_jsons):
+        txt_indices.append(int(txt.parts[-1].split('_')[0]))
```

Comment:
the same about extracting numbers. we know regex now - this time it will be cleaner and more robust solution

Reply:
done